### PR TITLE
Update build to work better as normal user

### PR DESCRIPTION
**Pull request by mjwolf**
Some small changes to the build process to work better when run in a container and as a normal user.

* Set Zig cache dir to be in the project binary dir
* Set libelf to install with current user permissions
* Remove fix-permissions build stage, as all files are now built with the current user as owner

### CODE REVIEW

**Review by rhysre — APPROVED** [review]
Thanks Mike!

File: Makefile [hunk 1]
```diff
@@ -167,8 +164,5 @@ endif
 	go install github.com/florianl/bluebox@b8590fb1850f56df6e6d7786931fcabdc1e9173d
 	cd testing && ./run_tests.sh ${ARCH} ${ARTIFACTS_PATH} ${PWD}/kernel-images/${IMG_FILTER}/${ARCH}/*
 
-fix-permissions:
-	${SUDO} chown -fR ${USER}:${USER} .
```
Comment (nicholasberlin) [inline Makefile:171]: :woohoo:

**Review by nicholasberlin — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -28,7 +28,7 @@ else
 endif
 
 ifdef NOCONTAINER
-	CONTAINER_RUN_CMD = 
+	CONTAINER_RUN_CMD =
 else
 	CONTAINER_RUN_CMD = ${CONTAINER_ENGINE} run --platform linux/${ARCH} --rm -v${PWD}:${PWD} -w${PWD} -e NOCONTAINER=TRUE ${CONTAINER_IMAGE}
```
Comment (lrishi) [inline Makefile:33]:
```suggestion
	CONTAINER_RUN_CMD = ${CONTAINER_ENGINE} run --platform linux/${ARCH} --rm -v${PWD}:${PWD} -w${PWD} -u$(id -u):$(id -g) -e NOCONTAINER=TRUE ${CONTAINER_IMAGE}
```

**Review by lrishi — CHANGES_REQUESTED** [review]
lgtm, blocking just to make sure one minor tweak is made.

File: Makefile (hunk 2 above)
Comment (mjwolf) [reply Makefile]: Done

**Review by lrishi — APPROVED** [review]
lgtm 👍🏽

**Pull request merged**